packages/puppeteer-core/src/common/JSHandle.ts

```diff
@@ -229,6 +229,8 @@ export class JSHandle<T = unknown> {
   }
 
   /**
+   * @deprecated Use `instanceof ElementHandle` instead of this method.
```

Comment:
does `instanceof ElementHandle` check maintain the ElementType generic? I think it should but not sure.

Reply:
I tested this, and it seems doing an `instanceof` check doesn't perform narrowing... So I am undeprecating this.